### PR TITLE
TR-5067 - order by documentation

### PR DESCRIPTION
**Pull request by ankrim**
This is how the page will look like:

Screen Shot 2019-12-04 at 12 19 11 PM

### CODE REVIEW

File: references/sql-operations.md [hunk 1]
```diff
@@ -26,22 +26,24 @@ SELECT <* or column selection or JSON template>
...
 
-The clauses `WHERE`, `EXPAND BY` and `LIMIT` can be used only if a `FROM` clause is used.
+The clauses `WHERE`, `EXPAND BY`, `ORDER BY` and `LIMIT` can be used only if a `FROM` clause is used.
 
 The clauses must appear in the order specified above.
 
 ### Order of execution
 
-1. `FROM`
+1. [`FROM`](/docs/references/sql-operations#from-clause)
```
Comment (ankrim) [inline references/sql-operations.md]: I wonder if this works and if it looks ok, I'll test

Comment (ankrim) [reply references/sql-operations.md]: It looks like this:

Screen Shot 2019-12-03 at 3 48 35 PM

I think it's ok

File: references/sql-operations.md [hunk 2]
```diff
@@ -1222,6 +1224,225 @@ Will create the following result set:
...
+WHERE <predicate>
+ORDER BY <path-1> [ASC|DESC], <path-2> [ASC|DESC], ..., <path-N> [ASC|DESC]
+```
+
+`<path>` describes the location of a value inside a JSON object or array. `<path>` contains one or more dot-separated keys/field names that describe the lookup chain of fields inside the input JSON object. See [Column selection](/docs/references/sql-operations#column-selection) for more details, or check the examples below.
+
+At least one `<path>` is required.
+The first path `<path-1>` will be the primary sort key, `<path-2>` will be the secondary sort key, etc.
+
+The ASC or DESC qualifiers are optional, the default will be ASC.
```
Comment (mooreds) [inline references/sql-operations.md]: Is it worth explaining that ASC is sorted ascending and DESC is sorted descending?

Also, is it numeric for numeric fields, lexicographic for string fields? How are boolean fields sorted?

I assume the path must point to a scalar value?

File: references/sql-operations.md [hunk 3]
```diff
@@ -1222,6 +1224,225 @@ Will create the following result set:
...
+
+`ORDER BY` is executed after `SELECT` - so any new field that was created in the `SELECT` clause can be used in the `ORDER BY` clause.
+
+`ORDER BY` is executed before `LIMIT` - therefore in order to get the correct set of rows after the sorting and limit we must get all the data and only after that do the sorting and limit. In other words - if the operation is paginated, a query with `ORDER BY` will have to paginate through all the results and might increase the number of API calls. You can use subqueries to create a specific order of execution.
```
Comment (mooreds) [inline references/sql-operations.md]: "through all the results" -> "through all the results up to the limit imposed by the API or the max number of rows set up for this data connection"

Is that correct?

Comment (ankrim) [reply references/sql-operations.md]: not exactly, we actually need to get all the data in order to make the sort correctly (for example if you compare this to a sql DB), so this is a problem if there is a lot of data and rate limits (the api rate limit and our rate limits).
for these extreme cases I mentioned the `You can use subqueries to create a specific order of execution.`.

(There might be a way to pass the sorting to the server (if it's supported) - that will be in the next blog post based on my talk in the api spec conference, but we're still far from implementing that)

File: references/sql-operations.md (hunk 2 above)
Comment (ankrim) [reply references/sql-operations.md]: done, thanks

File: references/sql-operations.md [hunk 4]
```diff
@@ -1222,6 +1224,225 @@ Will create the following result set:
...
+ORDER BY data.values[1] DESC
+```
+
+Will reorder the rows:
+
+```javascript
+[
+  {
+    "data": {
+      "values": [ 100, 200, 300 ]
```
Comment (blazarus) [inline references/sql-operations.md]: nit: maybe keep the formatting of these rows the same as above

File: references/sql-operations.md (hunk 3 above)
Comment (blazarus) [inline references/sql-operations.md]: maybe we should link to https://www.transposit.com/docs/references/faq/#automatic-pagination here?

**Pull request merged**